astronomer/providers/google/cloud/extractors/bigquery_extractor.py

```diff
@@ -0,0 +1,94 @@
+# SPDX-License-Identifier: Apache-2.0.
+
+import json
+import logging
+import traceback
+from typing import List, Optional
+
+import attr
+from openlineage.airflow.extractors.base import BaseExtractor, TaskMetadata
+from openlineage.airflow.utils import get_job_name
+from openlineage.client.facet import SqlJobFacet
+from openlineage.common.provider.bigquery import (
+    BigQueryDatasetsProvider,
+    BigQueryErrorRunFacet,
+)
+from openlineage.common.sql import SqlParser
+
+_BIGQUERY_CONN_URL = "bigquery"
+
+log = logging.getLogger(__name__)
+
+
+@attr.s
+class SqlContext:
+    """Internal SQL context for holding query parser results"""
+
+    sql: str = attr.ib()
+    inputs: Optional[str] = attr.ib(default=None)
+    outputs: Optional[str] = attr.ib(default=None)
+    parser_error: Optional[str] = attr.ib(default=None)
+
+
+class BigQueryExtractor(BaseExtractor):
+    def __init__(self, operator):
```

Comment:
add docstrings to explain what this class is supposed to do?

Reply:
done.